packages/slice-machine/src/domain/customType.ts

```diff
@@ -425,6 +453,64 @@ export function reorderField(args: ReorderFieldArgs): CustomType {
   return newCustomType;
 }
 
+// if the UID is not existing in any section, it should be added to the main section
+export function addUIDField(label: string, customType: CustomType): CustomType {
+  const sectionId = getMainSectionId(customType);
+  if (typeof sectionId === "undefined") {
+    return customType;
+  }
+  const newFieldId = "uid";
+  const newField: UID = {
+    config: {
+      label,
+    },
+    type: "UID",
+  };
+
+  return addField({ customType, sectionId, newField, newFieldId });
+}
+
+export function updateUIDField(
+  label: string,
+  customType: CustomType,
+): CustomType {
+  const [sectionId, section] = getSectionWithUIDField(customType) ?? [];
+
+  const fieldId = Object.keys(section ?? {})?.find(
```

Comment:
💡 #idea

I don't understand why you don't get the field ID from the next line as you have the field. It's really strange because you have a manual code here for the `fieldId` but for the `fieldValue` you have a function, it's not consistent.

It's coming back to the reason why we have for example `getMainSectionEntry`, by getting the entry you always ensure you get either the ID, the value or both.

What about having the same here?

Reply:
yes, that's good idea, will change that